### PR TITLE
More docs and samples

### PR DESCRIPTION
**Pull request by jasnell**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,9 +6,29 @@
...
+## Introduction
+
+Piscina works by creating a pool of Node.js Worker Threads to which
+one or more tasks may be dispatched. Each worker thread executes a
+single exported function defined in a separate file. Whenever a
+task is dispatched to a worker, the worker invokes the exported
+function and reports the return value back to Piscina when the
+function completes.
+
+## Current Limitations (Things we're working on / would love help with)
```
Comment (addaleax) [inline README.md:23]: Can we avoid putting this before the example? I feel somewhat strongly about an introductory example being one of the very first things in a README :)

File: README.md [hunk 2]
```diff
@@ -130,6 +220,22 @@ Is `true` if this code runs inside a `Piscina` threadpool as a Worker.
...
+within a Worker, the performance advantages of doing so will be
+minimal.
+
+Specifically, it is worth noting that asynchronous file system,
+network, and crypto operations within Node.js are already serviced
+by a thread pool at the libuv level, and that all Worker Threads
+created within a Node.js process share the same libuv thread pool.
+This means that there will be little performance impact on moving
+such async operations into a Piscina worker (see examples/scrypt
+for example).
```
Comment (addaleax) [inline README.md:237]:
```suggestion
Specifically, it is worth noting that asynchronous operations within Node.js,
including I/O such as file system operations or CPU-bound tasks such as 
crypto operations or compression algorithms, are already performed in
parallel by Node.js and libuv on a per-process level.
This means that there will be little performance impact on moving
such async operations into a Piscina worker (see examples/scrypt
for example).
```

File: examples/simple copy/index.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+const { resolve } = require('path');
+
+const piscina = new Piscina({
+  filename: resolve(__dirname, 'worker.js')
+});
+
+(async function() {
+  const result = await piscina.runTask({ a: 4, b: 6 });
+  console.log(result);  // Prints 10
+})();
```
Comment (addaleax) [inline examples/simple copy/index.js:13]: Is this file included intentionally?

File: examples/abort/index2.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+const Piscina = require('../../dist/src');
```
Comment (addaleax) [inline examples/abort/index2.js:3]:
```suggestion
const Piscina = require('../..');
```

File: examples/abort/index.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+const Piscina = require('../../dist/src');
```
Comment (addaleax) [inline examples/abort/index.js:3]:
```suggestion
const Piscina = require('../..');
```

File: examples/abort/worker.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+const { promisify } = require('util');
```
Comment (addaleax) [inline examples/abort/worker.js:1]:
```suggestion
'use strict';
const { promisify } = require('util');
```

File: examples/simple_async/index.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+const Piscina = require('../../dist/src');
```
Comment (addaleax) [inline examples/simple_async/index.js:3]:
```suggestion
const Piscina = require('../..');
```

File: examples/simple_async/worker.js [hunk 8]
```diff
@@ -0,0 +1,8 @@
+const { promisify } = require('util');
```
Comment (addaleax) [inline examples/simple_async/worker.js:1]:
```suggestion
'use strict';
const { promisify } = require('util');
```

File: examples/simple/worker.js [hunk 9]
```diff
@@ -0,0 +1,3 @@
+module.exports = ({ a, b }) => {
```
Comment (addaleax) [inline examples/simple/worker.js:1]:
```suggestion
'use strict';
module.exports = ({ a, b }) => {
```

File: README.md (hunk 1 above)
Comment (jasnell) [reply README.md:23]: Missed these :-) ... will make these edits now

File: examples/simple copy/index.js (hunk 3 above)
Comment (jasnell) [reply examples/simple copy/index.js:13]: ugh. no, thought this had been deleted

**Pull request merged**